static/app/utils/withCustomerDomain.spec.tsx

```diff
@@ -0,0 +1,311 @@
+      {context: routerContext}
+    );
+
+    expect(container).toBeEmptyDOMElement();
+    expect(router.replace).toHaveBeenCalledTimes(1);
+    expect(router.replace).toHaveBeenCalledWith('/settings/react/alerts/?q=123#hash');
+    expect(spyWithScope).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not redirect when :orgId is not present in the routes', function () {
+    const organization = TestStubs.Organization({
+      slug: 'albertos-apples',
+      features: ['customer-domains'],
+    });
+
+    const params = {};
+
+    const {router, route, routerContext} = initializeOrg({
+      ...initializeOrg(),
+      organization,
+      router: {
+        params,
+        // /settings/account/notifications/reports/
+        routes: [
+          {path: '/', childRoutes: []},
+          {childRoutes: []},
+          {path: '/settings/', name: 'Settings', indexRoute: {}, childRoutes: []},
+          {name: 'Account', path: 'account/', childRoutes: []},
+          {name: 'Notifications', path: 'notifications/', childRoutes: []},
+          {name: 'Reports', path: 'reports/'},
+        ],
+      },
+    });
+
+    const MyComponentWithApi = withCustomerDomain(MyComponent);
+    render(
+      <OrganizationContext.Provider value={organization}>
+        <MyComponentWithApi
+          router={router}
+          location={router.location}
+          params={params}
+          routes={router.routes}
+          routeParams={router.params}
+          route={route}
+        />
+      </OrganizationContext.Provider>,
+      {context: routerContext}
+    );
+
+    expect(screen.getByText('Org slug: no org slug')).toBeInTheDocument();
+    expect(router.replace).not.toHaveBeenCalled();
+    expect(spyWithScope).not.toHaveBeenCalled();
+  });
+
+  it('redirect option that provides alternative redirect destination', function () {
+    const organization = TestStubs.Organization({
+      slug: 'albertos-apples',
+      features: ['customer-domains'],
+    });
+
+    const params = {
+      orgId: organization.slug,
+      projectId: 'react',
+    };
+    const {router, route, routerContext} = initializeOrg({
+      ...initializeOrg(),
+      organization,
+      router: {
+        params,
+        routes: [
+          // /settings/:orgId/
+          {path: '/', childRoutes: []},
+          {childRoutes: []},
+          {path: '/settings/', name: 'Settings', indexRoute: {}, childRoutes: []},
+          {name: 'Organizations', path: ':orgId/', childRoutes: []},
+        ],
+      },
+    });
+
+    const MyComponentWithApi = withCustomerDomain(MyComponent, {
+      redirect: [
+        {
+          // If /settings/:orgId/ is encountered, then redirect to /settings/organization/ rather than redirecting to
+          // /settings/.
+          from: '/settings/:orgId/',
+          to: '/settings/organization/',
+        },
+      ],
+    });
+    const {container} = render(
+      <OrganizationContext.Provider value={organization}>
+        <MyComponentWithApi
```

Comment:
```suggestion
    const WrappedComponent = withCustomerDomain(MyComponent, {
      redirect: [
        {
          // If /settings/:orgId/ is encountered, then redirect to /settings/organization/ rather than redirecting to
          // /settings/.
          from: '/settings/:orgId/',
          to: '/settings/organization/',
        },
      ],
    });
    const {container} = render(
      <OrganizationContext.Provider value={organization}>
        <WrappedComponent
```